test/contract_tests.js

```diff
@@ -0,0 +1,543 @@
+        Einstance = await upgrades.deployProxy(Escrow, [instance.address], { unsafeAllow: ['delegatecall'] });
+
+        // Adding to `ERC20Payer` address to TestToken SC `Allowance` so that `ERC20Payer`can pay in ERC20
+        await tkk_instance.allowance(ERC20Payer.address, Einstance.address);
+        await tkk_instance.approve(Einstance.address, ethers.utils.parseUnits("100000", "ether")); // ether format for 100000 tokens
+
+        /*
+        console.log(ethers.utils.formatEther(await tkk_instance.allowance(ERC20Payer.address, instance.address))) // returns the allowance (uint256)amount
+        console.log("ERC20Payer :", await ERC20Payer.address, "Client :", await Client.address);
+        console.log("TTKN_address : ", await tkk_instance.address, "ERC20Payer BALANCE :", ethers.utils.formatEther(await tkk_instance.balanceOf(ERC20Payer.address)))
+        */
+
+        // Constant Reports
+        salt = generateRandomSalt();
+
+        report1 = { "id": 1, "salts": salt, "paymentWalletAddress": Reporter1.address, "project": "0xbf971d4360414c84ea06783d0ea51e69035ee526" }
+        report2 = { "id": 2, "salts": salt, "paymentWalletAddress": Reporter2.address, "project": "0xfbb1b73c4f0bda4f67dca266ce6ef42f520fbb98" }
+
+        // `Deployer` is the only operator on deployProxy,To keep Deployer independent for proxy upgrades tasks \ 
+        // Assigning `OPERATOR` role to `Triager` who will be responsible for all `OPERATOR` operations on `Notary`contract functions.
+        await instance.connect(Deployer).grantRole(operator_keccak, Triager.address);
+    });
+
+    it("precheck: if `Escrow` and `Notary` is on same proxy interface.",async function(){
+        let escrow_notary_proxy_addr = await Einstance.connect(Deployer).notary(); // get the address of notary interface from Escrow.
+        await expect(escrow_notary_proxy_addr).to.be.equal(instance.address);
+    })
+
+    describe("===> submit()", function () {
+        let getReportRoot;
+
+        beforeEach(async () => {
+            getReportRoot = F_SUBMIT(report1);
+        });
+
+        it("submit(): Only Callable by the operator role only", async function () {
+            let resp = await instance.connect(Triager).submit(getReportRoot);
+            let block = await ethers.provider.getBlock(resp.blockNumber);
+            expect(resp).to.emit(instance, "ReportSubmitted").withArgs(getReportRoot, block.timestamp);
+        })
+
+        it("submit(): Revert if called by other than OPERATOR ROLE", async function () {
+            await expect(instance.connect(Reporter1).submit(getReportRoot)).to.be.reverted;
+        })
+
+        it("submit(): Only Accepts a report root format of merkle tree constructed from the report", async function () {
+            //  accept
+            await expect(instance.connect(Triager).submit(random_bytes_RAND));
+            //  reject
+            await expect(instance.connect(Triager).submit(address_ZERO)).to.be.reverted;
```

Comment:
Since `submit()` accepts the report root of `bytes32`

1. passing `all-zeroes bytes32` i.e `ethers.utils.formatBytes32String(0)` is allowed, Whereas
2. passing non `bytes32` i.e of type `address` should revert.

What should i add to the current test case?